python_modules/dagster/dagster/_core/events/__init__.py

```diff
@@ -1768,69 +1716,77 @@ def __new__(cls, file_key, step_keys, external_url=None):
 ###################################################################################################
 
 
-# Keep these around to prevent issues like https://github.com/dagster-io/dagster/issues/3533
-@whitelist_for_serdes
-class AssetStoreOperationData(NamedTuple):
-    op: str
-    step_key: str
-    output_name: str
-    asset_store_key: str
-
-
-@whitelist_for_serdes
-class AssetStoreOperationType(Enum):
-    SET_ASSET = "SET_ASSET"
-    GET_ASSET = "GET_ASSET"
-
-
-@whitelist_for_serdes
-class PipelineInitFailureData(NamedTuple):
-    error: SerializableErrorInfo
```

Comment:
are these under test? confident that theres no issue here?

Reply:
So, I _thought_ these were under test (certainly a lot of other backcompat stuff is) but on further investigation it appears maybe not? Deleting these did not break any core/storage/general tests in `master`, and grepping for `AssetStoreOperation{Data,Type}` doesn't turn up anything (like a dict with `"__class__": "AssetStoreOperationType"`).

But IIUC:

- these only appear in serialized events
- the `handle_backcompat` function has logic to convert these before they are unpacked
- The `DagsterEventSerializer` uses a fallback `handle_unpack_error` to just return a generic event with an error message for any event that fails to unpack (this functionality was already in place in the status quo)

So I'm pretty confident these are safe, it looks like they were superfluous even before this PR because of the generic event returned for events that fail to unpack (and that's definitely under test in `core_tests/test_event_logging.py`).